protocol/shannon/protocol.go

```diff
@@ -180,60 +180,35 @@ func (p *Protocol) AvailableEndpoints(
 	return endpointAddrs, buildSuccessfulEndpointLookupObservation(serviceID), nil
 }
 
-// BuildRequestContextForEndpoint creates a new protocol request context for a specified service and endpoint.
+// BuildHTTPRequestContextForEndpoint creates a new HTTP protocol request context for a specified service and endpoint.
 //
 // Parameters:
 //   - ctx: Context for cancellation, deadlines, and logging.
 //   - serviceID: The unique identifier of the target service.
 //   - selectedEndpointAddr: The address of the endpoint to use for the request.
 //   - httpReq: ONLY used in Delegated mode to extract the selected app from headers.
-//   - TODO_TECHDEBT: Decouple context building for different gateway modes.
 //
 // Behavior:
-//   - Retrieves active sessions for the given service ID from the full node.
-//   - Retrieves unique endpoints available across all active sessions
-//   - Filtering out sanctioned endpoints from list of unique endpoints.
+//   - Selects the endpoint that matches the pre-selected address.
 //   - Obtains the relay request signer appropriate for the current gateway mode.
-//   - Returns a fully initialized request context for use in downstream protocol operations.
+//   - Returns a fully initialized HTTP request context for use in downstream protocol operations.
 //   - On failure, logs the error, returns a context setup observation, and a non-nil error.
-//
-// Implements the gateway.Protocol interface.
-func (p *Protocol) BuildRequestContextForEndpoint(
+func (p *Protocol) BuildHTTPRequestContextForEndpoint(
 	ctx context.Context,
 	serviceID protocol.ServiceID,
 	selectedEndpointAddr protocol.EndpointAddr,
 	httpReq *http.Request,
 ) (gateway.ProtocolRequestContext, protocolobservations.Observations, error) {
 	logger := p.logger.With(
-		"method", "BuildRequestContextForEndpoint",
 		"service_id", serviceID,
 		"endpoint_addr", selectedEndpointAddr,
 	)
 
-	activeSessions, err := p.getActiveGatewaySessions(ctx, serviceID, httpReq)
-	if err != nil {
-		logger.Error().Err(err).Msgf("Relay request will fail due to error retrieving active sessions for service %s", serviceID)
-		return nil, buildProtocolContextSetupErrorObservation(serviceID, err), err
-	}
-
-	// Retrieve the list of endpoints (i.e. backend service URLs by external operators)
-	// that can service RPC requests for the given service ID for the given apps.
-	// This includes fallback logic if session endpoints are unavailable.
-	// The final boolean parameter sets whether to filter out sanctioned endpoints.
-	endpoints, err := p.getUniqueEndpoints(ctx, serviceID, activeSessions, true)
+	// Retrieve the endpoint that matches the pre-selected address.
```

Comment:
BLOCKER: please do not make any changes to this method, other than renaming it, in this PR.